### ISSUE TITLE
React stateless component + children : error Property 'children' is missing in type 'IntrinsicAttributes

### ISSUE THREAD

**Issue by Offirmo** [Microsoft/TypeScript#8588]
[x] Many common issues and suggestions are addressed in the FAQ
     https://github.com/Microsoft/TypeScript/wiki/FAQ
[x] Search for duplicates before logging new issues
    https://github.com/Microsoft/TypeScript/issues?utf8=%E2%9C%93&q=is%3Aissue
[x] Questions are best asked and answered at Stack Overflow
    http://stackoverflow.com/questions/tagged/typescript

---

-> asked on S/O here http://stackoverflow.com/questions/37189910/react-stateless-component-children-with-typescript-error-property-children no responses so far.

**TypeScript Version:** 
1.8.2

**Code**

``` tsx
// presentational components
const MsgDisplay = ({msg}) => (
  <div>
    {msg} // will be more complicated later
  </div>)

const RowDisplay = ({data, children}) => (
  <tr>
    <td>...</td>
    <td>{ children }</td>
    <td>...</td>
  </tr>)

// container component
public render() {
    return (
      <RowDisplay data={this.props.xyz} >
        <MsgDisplay msg={someMsg} />
      </RowDisplay>
    )
  }
```

**Expected behavior:**

Should compile.

**Actual behavior:** 

Complains that `error TS2324: Property 'children' is missing in type 'IntrinsicAttributes & { data: any; children: any; }'.`

Seems it can't see the children being passed in jsx notation.

**Comment by JabX** [comment]
I also face this problem, but I go around it by making children optional. Not a good solution though.

**Comment by yuit** [comment]
I think this issue is similar to #7992 and #6471

**Comment by donaldpipowitch** [comment]
Nice to see that this is planned for 2.1. It makes working with React really hard currently.

**Comment by Igorbek** [comment]
Does it progress any further? Is it going to be covered by #13618?
I just recently ran to this issue. It's pretty annoying when you components constructed as HOCs.

**Comment by mhegazy** [comment]
This looks like a duplicate of https://github.com/Microsoft/TypeScript/issues/13618, we would like to get that addressed in TS 2.3.

**Comment by mhegazy** [comment]
Closing in favor of https://github.com/Microsoft/TypeScript/issues/13618
